libs/network/network.h

```diff
@@ -72,17 +72,17 @@ typedef struct JsNetwork {
   bool (*checkError)(struct JsNetwork *net);
 
   /// if host=0, creates a server otherwise creates a client (and automatically connects). Returns >=0 on success
-  int (*createsocket)(struct JsNetwork *net, uint32_t host, unsigned short port, SocketType socketType);
+  int (*createsocket)(struct JsNetwork *net, uint32_t host, unsigned short port, SocketType socketType, JsVar *options);
   /// destroys the given socket
   void (*closesocket)(struct JsNetwork *net, int sckt);
   /// If the given server socket can accept a connection, return it (or return < 0)
   int (*accept)(struct JsNetwork *net, int sckt);
   /// Get an IP address from a name
   void (*gethostbyname)(struct JsNetwork *net, char * hostName, uint32_t* out_ip_addr);
   /// Receive data if possible. returns nBytes on success, 0 on no data, or -1 on failure
-  int (*recv)(struct JsNetwork *net, int sckt, void *buf, size_t len);
+  int (*recv)(struct JsNetwork *net, int sckt, void *buf, size_t len, uint32_t *host, unsigned short *port);
```

Comment:
Passing host/port to everything feels a bit ugly... I guess it works, but somewhere else you mentioned the need to be able to receive multiple packets?

What if the data returned/sent for UDP was just an array of bytes: 

`[PKTLEN_HI, PKTLEN_LO, PORT_HI, PORT_LO, HOST4, HOST3, HOST2, HOST1, packet data ... ]`

Just a thought - but it'd really reduce the amount of changes needed throughout the code?